scripts/buildorder.py

```diff
@@ -347,19 +329,63 @@ def generate_target_buildorder(target_path, pkgs_map, fast_build_mode):
         package.deps.difference_update([subpkg.name for subpkg in package.subpkgs])
     return package.recursive_dependencies(pkgs_map)
 
+def get_list_cyclic_dependencies(pkgs_map, index=None, checked=None, pkgname=None, return_list=False):
+    "Find and print (or return a list of) circular dependencies for all packages or for one specified package."
+    # is_root - indicates that the function was run for the first time (called root function)
```

Comment:
I've never heard this term before. This is usually called a "base case": https://en.wikipedia.org/wiki/Recursion#base_case

The comment helps but I don't think there's a reason not to use the common term, which doesn't require one, here.